nucypher/characters/lawful.py

```diff
@@ -1727,6 +1806,37 @@ def interface_info_with_metadata(self):
     # Re-Encryption
     #
 
+    def verify_kfrag_authorization(self,
+                                   alice: Alice,
+                                   kfrag: KFrag,
+                                   signed_writ: bytes,
+                                   work_order: 'WorkOrder'
+                                   ) -> None:
+        from nucypher.policy.orders import WorkOrder  # TODO: resolve ciruclar dependency
+
+        writ_hrac, writ_kfrag_checksum, writ_signature = WorkOrder.signed_writ_splitter(signed_writ)
+        reconstructed_writ = writ_hrac + writ_kfrag_checksum
+
+        try:
+            self.verify_from(alice, reconstructed_writ, signature=writ_signature)
+        except InvalidSignature:
+            raise Policy.Unauthorized  # This isn't from Alice (relayer).
+
+        if writ_hrac != work_order.hrac:  # Funky Workorder
+            raise Policy.Unauthorized  # Bob, what the *hell* are you doing?
+
+        kfrag_checksum = keccak_digest(bytes(kfrag))[:WRIT_CHECKSUM_SIZE]
```

Comment:
Any particular reason we're using a separate constant here instead of `KECCAK_DIGEST_LENGTH`?